waftools/detections/compiler_swift.py

```diff
@@ -2,11 +2,9 @@
 
 def __run(cmd):
     try:
-        cmd = Utils.subprocess.Popen(cmd,
-                                     stdout=Utils.subprocess.PIPE,
-                                     stderr=Utils.subprocess.PIPE,
-                                     shell=True)
-        output = cmd.stdout.read().decode().strip()
+        cmd = Utils.subprocess.Popen(cmd, stdout=Utils.subprocess.PIPE,
+                                     universal_newlines=True, shell=True)
```

Comment:
probably should have brought this up earlier, but why does it need `shell=True`?

Reply:
because i pass a single string as command and not an argument list. is there a problem with `shell=True` and would your prefer to not use it?

Reply:
I guess it's not the end of the world, since this is only getting trusted input (though maybe someone could abuse  $MACOS_SDK?), the main thing is this starts an additional program, and one that may vary in unpredictable ways depending on user configuration. 

https://stackoverflow.com/questions/3172470/actual-meaning-of-shell-true-in-subprocess

Either way, I think you might be able to use `subprocess.check_output` here.